File: iOS/TUILiveKit/Source/Common/UIComponent/LiveList/Store/LiveListState.swift
//
//  LiveListState.swift
//  TUILiveKit
//
//  Created by adamsfliu on 2024/5/30.
//

import Foundation
import RTCRoomEngine

struct LiveListResult {
    var cursor: String = ""
    var liveInfoList: [TUILiveInfo] = []
}

extension LiveListResult: Equatable {}

struct LiveListState {
    var liveInfoListResult: LiveListResult = LiveListResult()
}

struct LiveListNavigationState {
    enum Router {
        case exit
        case main
        case toLive(_ liveInfo: TUILiveInfo)
    }
    var currentRouter: Router = .main
}

extension LiveListNavigationState.Router: Equatable {
    static func ==(lhs: LiveListNavigationState.Router, rhs: LiveListNavigationState.Router) -> Bool {
        switch (lhs, rhs) {
        case (.exit, .exit),
            (.main, .main):
            return true
        case let (.toLive(l), .toLive(r)):
            return l == r
        case (.exit, _),
            (.main, _),
            (.toLive,_):
            return false
        }
    }
}

File: Flutter/livekit/lib/live_stream/features/live_list/store/live_list_state.dart
import 'package:flutter/material.dart';
import 'package:rtc_room_engine/rtc_room_engine.dart';

class LiveListState {
  final ValueNotifier<List<TUILiveInfo>> liveInfoList =
      ValueNotifier<List<TUILiveInfo>>([]);
  final ValueNotifier<bool> refreshStatus = ValueNotifier<bool>(false);
  final ValueNotifier<bool> loadStatus = ValueNotifier<bool>(false);
  final ValueNotifier<bool> isHaveMoreData = ValueNotifier<bool>(false);
  String cursor = "";
}
